app/services/user/bulk_destroy.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+class User::BulkDestroy
+  include Service::Base
+
+  params do
+    attribute :user_ids, :array
+
+    validates :user_ids, length: { maximum: 100 }
+  end
+
+  model :users
+  policy :can_delete_users
+  step :delete
+
+  private
+
+  def fetch_users(params:)
+    ids = params.user_ids
+    # this order cluase ensures we retrieve the users in the same order as the
+    # IDs in the param. we do this to ensure the users are deleted in the same
+    # order as they're selected in the UI
+    User.where(id: ids).order(DB.sql_fragment("array_position(ARRAY[?], users.id)", ids))
+  end
+
+  def can_delete_users(guardian:, users:)
```

Comment:
Would normally expect a `?` at the end of the method name here.

Reply:
Yeah if this was a method that we call directly, then a `?` at the end would be appropriate. However, this method name is determined by the `policy` call in the service class definition, and I think `policy :can_delete_users?` doesn't read too well and in all other services we don't use this naming pattern.